File: libs/angular/data-access/src/lib/models/installer-properties.ts
export interface InstallerProperties {
    installationLocation: string; // install location
    productName: string; // app name
    icon: string; // icon
    productVersion: string; // version
    publisher: string; // publisher
    supportLink: string; // support link
    supportEmail: string; // support email
    comment: string; // comment
    launchFile: string; // exe file
    runAsAdmin: boolean; // run as administrator
    launchApp: boolean; // run app after install
    shortcutInDesktop: ShortcutInDesktop
}

export interface ShortcutInDesktop {
    isCreated: boolean;
    runAsAdmin: boolean;
    shortcutName: string;
}

File: libs/rust/domain/src/models/installer_properties.rs
use serde::{Deserialize, Serialize};

#[derive(Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct InstallerProperties {
    pub installation_location: String, // install location
    pub product_name: String,          // app name
    pub icon: String,                  // icon
    pub product_version: String,       // version
    pub publisher: String,             // publisher
    pub support_link: String,          // support link
    pub support_email: String,         // support email
    pub comment: String,               // comment
    pub launch_file: String,           // exe file
    pub run_as_admin: bool,            // run as administrator
    pub launch_app: bool,              // run app after install
    pub shortcut_in_desktop: ShortcutInDesktop,
}

#[derive(Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ShortcutInDesktop {
    pub is_created: bool,
    pub run_as_admin: bool,
    pub shortcut_name: String,
}

